instiki2git.py

```diff
@@ -164,7 +164,8 @@ def read_latest_revision_id(latest_revision_file):
 def load_and_commit_new_revisions(repo_path, db_config, web_id, \
                                   latest_revision_file):
   latest_revision_id = read_latest_revision_id(latest_revision_file)
-  revs = load_new_revisions(db_config, web_id, latest_revision_id)
+  twentyseven_hours_ago = datetime.datetime.now() - datetime.timedelta(hours=27)
+  revs = load_new_revisions_before(db_config, web_id, latest_revision_id, twentyseven_hours_ago)
```

Comment:
I think this can cause revisions to be lost.

Suppose there is:
* a revision with id N at time T + 20 min
* a revision with id N + 1 at time T.

This can happen when revision N is updated (within the 30 minutes window) after revision N + 1 is created.

Suppose the script runs at time T + 27 h + 10 min. It will not include revision N, but will include revision N + 1. However, it will remember N + 1 as the last revision id for the next time it runs, effectively skipping revision N.

Reply:
One solution could be to go by `updated_at` exclusively.

Reply:
Wow - thanks for the catch ;-)

Reply:
> One solution could be to go by updated_at exclusively.

But this is only possible if the clock is monotonous...

Reply:
Do you know (or is it documented somewhere), what the times ```created_at```, ```updated_at``` and ```revised_at``` acutally mean? I would have guessed the first is the creation time of the page (therefore earlier than the others), the second one is the time the latest version of the revision went live and the last one is the time of the latest update of the revision. But there are revisions, where ```created_at``` is later than ```revised_at```...

Reply:
![grafik](https://user-images.githubusercontent.com/22154668/195570305-fa74682e-727f-4d9b-930a-64381f515fae.png)

Reply:
* `created_at` and `updated_at` are automatically written by Rails when the entry is created or updated.
* `revised_at` is managed by nlab-rails. It is when the `revise` method of the page object is called. This is before the database entry is actually modified. The difference is basically the time it takes to render the page once (to confirm the edit gets accepted).

Reply:
Thanks!

Reply:
Ich habe mal eine Variante gebaut, die einen Zeitbereich nimmt.